test/lint/lint-filenames-permissions.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""
+This checks that all files in the repository have correct filenames and permissions
+"""
+
+import os
+import re
+import sys
+from subprocess import check_output
+
+CMD_ALL_FILES = "git ls-files --full-name"
+CMD_SOURCE_FILES = 'git ls-files --full-name -- "*.[cC][pP][pP]" "*.[hH]" "*.[pP][yY]" "*.[sS][hH]"'
+CMD_SHEBANG_FILES = """git grep --full-name --line-number -I '^#!' | grep :1: | cut -d ":" -f 1"""
```

Comment:
Nit: Consider doing the filtering (`grep`, `cut`) in Python instead to make the script easier to read and maintain. In the Python code a comment could be added that `:1:` is used to make sure only shebangs on the first line are considered.

Reply:
updated to move the filtering into python :)